tests/translator/output/api_with_cors_and_conditions_no_definitionbody.json

```diff
@@ -0,0 +1,381 @@
+                  "uri": {
+                    "Fn::Sub": "arn:aws:apigateway:${AWS::Region}:lambda:path/2015-03-31/functions/${ImplicitApiFunction2.Arn}/invocations"
+                  }
+                },
+                "responses": {}
+              },
+              "get": {
+                "Fn::If": [
+                  "MyCondition",
+                  {
+                    "x-amazon-apigateway-integration": {
+                      "type": "aws_proxy",
+                      "httpMethod": "POST",
+                      "uri": {
+                        "Fn::If": [
+                          "MyCondition",
+                          {
+                            "Fn::Sub": "arn:aws:apigateway:${AWS::Region}:lambda:path/2015-03-31/functions/${ImplicitApiFunction.Arn}/invocations"
+                          },
+                          {
+                            "Ref": "AWS::NoValue"
+                          }
+                        ]
+                      }
+                    },
+                    "responses": {}
+                  },
+                  {
+                    "Ref": "AWS::NoValue"
+                  }
+                ]
+              },
+              "post": {
+                "Fn::If": [
+                  "MyCondition",
+                  {
+                    "x-amazon-apigateway-integration": {
+                      "type": "aws_proxy",
+                      "httpMethod": "POST",
+                      "uri": {
+                        "Fn::If": [
+                          "MyCondition",
+                          {
+                            "Fn::Sub": "arn:aws:apigateway:${AWS::Region}:lambda:path/2015-03-31/functions/${ImplicitApiFunction.Arn}/invocations"
+                          },
+                          {
+                            "Ref": "AWS::NoValue"
+                          }
+                        ]
+                      }
+                    },
+                    "responses": {}
+                  },
+                  {
+                    "Ref": "AWS::NoValue"
+                  }
+                ]
+              },
+              "options": {
+                "summary": "CORS support",
+                "consumes": [
+                  "application/json"
+                ],
+                "produces": [
+                  "application/json"
+                ],
+                "x-amazon-apigateway-integration": {
+                  "type": "mock",
+                  "requestTemplates": {
+                    "application/json": "{\n  \"statusCode\" : 200\n}\n"
+                  },
+                  "responses": {
+                    "default": {
+                      "statusCode": "200",
+                      "responseParameters": {
+                        "method.response.header.Access-Control-Allow-Origin": "'www.example.com'",
+                        "method.response.header.Access-Control-Allow-Methods": "'DELETE,GET,OPTIONS,POST'"
```

Comment:
DELETE method doesn't have a condition on it, while the other 2 do. I verified that this can still be deployed to CFN if the other methods (POST, GET) are not included in the path (condition = false) but included here in CORS. Is it fine if only the DELETE method exists, but CORS allows GET and POST as well (which are removed by CFN by resolving conditions)?

Reply:
This is fine.